recipes-connectivity/avahi/avahi_%.bbappend

```diff
@@ -0,0 +1,6 @@
+do_install:append() {
+    # On Torizon we use avahi-daemon to setup the hostname. So we need to properly
+    # order it's service to only run after 'set-hostname.service' finishes.
+    sed -i '/^After=systemd-resolved.service connman.service/ s/$/ network-pre.target/' ${D}${systemd_system_unitdir}/avahi-daemon.service
+    sed -i '/^Description=/a Wants=network-pre.target' ${D}${systemd_system_unitdir}/avahi-daemon.service
```

Comment:
Can we assume GNU `sed` is always used in Yocto? I'm asking because the POSIX way of the `sed` "append" command requires a backslash and a newline as described in https://pubs.opengroup.org/onlinepubs/9699919799/utilities/sed.html:

```
[1addr]a\
text
    Write text to standard output as described previously.
```

Looking at existing OE scripts I see there are some cases where it's assumed the newline is not needed like here: https://github.com/openembedded/openembedded-core/blob/master/meta/recipes-core/initscripts/initscripts_1.0.bb#L110

So, if we follow the same assumption we should at least add the backslash:

```
sed -i '/^Description=/a\Wants=network-pre.target' ${D}${systemd_system_unitdir}/avahi-daemon.service
```

which works with GNU `sed` even if you pass the `--posix` switch. On the other hand, not passing the backslash does not work when `--posix` is passed to GNU `sed` which confirms the fact that the standard requires it.

Reply:
Ah, nice to know!
I've updated the commit.